contrib/precompile_generate.jl

```diff
@@ -0,0 +1,96 @@
+function gen_single_tracefile(
+        code::AbstractString,
+        tracefile::AbstractString,
+    )
+    julia_binary = Base.julia_cmd().exec[1]
+    cmd = `$(julia_binary)`
+    push!(cmd.exec, "--compile=all")
+    push!(cmd.exec, "--trace-compile=$(tracefile)")
+    push!(cmd.exec, "-e $(code)")
+    splitter = Sys.iswindows() ? ';' : ':'
+    project = Base.active_project()
+    env2 = copy(ENV)
+    env2["JULIA_LOAD_PATH"] = "$(project)$(splitter)@stdlib"
+    env2["JULIA_PROJECT"] = "$(project)"
+    run(setenv(cmd, env2))
+    return nothing
+end
+
+function gen_single_precompile(code::AbstractString)
+    str = mktempdir() do dir
+        tracefile = joinpath(dir, "tracefile")
+        gen_single_tracefile(code, tracefile)
+        return read(tracefile, String)::String
+    end::String
+    lines = convert(
+        Vector{String},
+        strip.(split(strip(str), '\n')),
+    )::Vector{String}
+    filter!(line -> !isempty(line), lines)
+    return lines
+end
+
+function gen_all_precompile()
+    codes = String[
+        "import HTTP; HTTP.get(\"https://example.com/\")",
+    ]
+    all_lines = String[]
+    for code in codes
+        lines = gen_single_precompile(code)
+        append!(all_lines, lines)
+    end
+    unique!(all_lines)
+end
+
+function write_all_precompile(io::IO, all_lines::Vector{String})
+    preamble_lines = String[
+        "import MbedTLS",
+        "const MbedTLS_jll = MbedTLS.MbedTLS_jll"
+    ]
+    for line in preamble_lines
+        println(io, line)
+    end
+    println(io)
+    println(io, """
+    function _precompile()
+    """)
+    println(io, """
+        if ccall(:jl_generating_output, Cint, ()) != 1
+            return nothing
+        end
+    """)
+    println(io, """
+        @static if Base.VERSION < v"1.9-"
```

Comment:
This is great, thanks for the PR @DilumAluthge ! I am surprised that this is only enabled for Julia 1.9 and up. [We found](https://github.com/fonsp/Pluto.jl/pull/1934) that adding `precompile` calls also gave improvements in Julia 1.8.

Reply:
@DilumAluthge why this is only enabled for 1.9 and later?

Reply:
So, I was under the impression that without https://github.com/JuliaLang/julia/pull/43990, we can't cache `CodeInstances` from other packages, and thus the benefit of this would be far less noticeable.